opm/autodiff/ParallelOverlappingILU0.hpp

```diff
@@ -442,17 +442,30 @@ namespace Opm
         upper.resize( A.N() );
         inv.resize( A.N() );
 
-        lower.reserveAdditional( 2*A.N() );
+        // Count the lower and upper matrix entries.
+        size_type numLower = 0;
+        size_type numUpper = 0;
+        const auto endi = A.end();
+        for (auto i=A.begin(); i!=endi; ++i) {
+          const size_type iIndex = i.index();
+          for (auto j = (*i).begin(); j.index() < iIndex; ++j) {
+              ++numLower;
+          }
+          for (auto j = (*i).beforeEnd(); j.index() > iIndex; --j) {
+              ++numUpper;
+          }
```

Comment:
Same here.

Reply:
> Actually, the correct number of columns does not have to be computed, because the entries are later added with push_back. The reserve is just to make sure that not to many memory re-allocations have to be done.

I realize that it is just an optimization, and that it does not have to be precise, as long as it does not underestimate. The original code underestimated, and reserved more inside the loop.

> BTW, iterating over the column backwards is much slower than iterating in the normal order, which is also the way these entries appear in memory.

If so, why is it done in the original code? I did not understand why, but decided to copy the pattern just to be safe.